### ISSUE TITLE
panic: Milestone should never be updated

### ISSUE THREAD

**Issue by Thoralf-M** [iotaledger/hornet#590]
- HORNET version HORNET-0.5.0-rc7_Windows_x86_64

```
panic: Milestone should never be updated: BLRFPIRZAUNRADRNKLWSQEAEECLSKWWUXKEQUMYORHRIKPDPAGJTXBLAZVUFQOYCWGDREMGNWIYLDD999 (648401)
 goroutine 289 [running]:
 github.com/gohornet/hornet/pkg/model/tangle.(*Milestone).Update(0xc001975800, 0x2232860, 0xc004a92de0)
/__w/hornet/hornet/pkg/model/tangle/milestones_storage.go:66 +0x11e
 github.com/iotaledger/hive.go/objectstorage.(*CachedObjectImpl).updateResult(0xc00ddbbc80, 0x2232860, 0xc004a92de0)
/github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/objectstorage/cached_object.go:162 +0x210 
github.com/iotaledger/hive.go/objectstorage.(*ObjectStorage).putObjectInCache(0xc000156980, 0x2232860, 0xc004a92de0, 0x0)
/github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/objectstorage/object_storage.go:730 +0xb3 
github.com/iotaledger/hive.go/objectstorage.(*ObjectStorage).Store(0xc000156980, 0x2232860, 0xc004a92de0, 0xc00af29a70, 0xa1a549)
/github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/objectstorage/object_storage.go:70 +0xb3 
github.com/gohornet/hornet/pkg/model/tangle.StoreMilestone(0xc00420f0e0, 0x2232701)
 /__w/hornet/hornet/pkg/model/tangle/milestones_storage.go:146 +0xdc
 github.com/gohornet/hornet/pkg/model/tangle.tryConstructBundle(0xc00a5943a0, 0xc00a594300)
/__w/hornet/hornet/pkg/model/tangle/bundle_storage.go:402 +0x95e
 github.com/gohornet/hornet/pkg/model/tangle.tryConstructBundle(0xc00a66c320, 0xc00a66c300)
/__w/hornet/hornet/pkg/model/tangle/bundle_storage.go:329 +0x2d2
 github.com/gohornet/hornet/pkg/model/tangle.AddTransactionToStorage(0xc00dea3d40, 0x1010009e55e, 0x1, 0x83dd7fa934dc0)
/__w/hornet/hornet/pkg/model/tangle/bundle_storage.go:304 +0x7a7
 github.com/gohornet/hornet/plugins/tangle.processIncomingTx(0xc00dea3d40, 0xc00a433040, 0xc000e00a50)
/__w/hornet/hornet/plugins/tangle/tangle_processor.go:129 +0x9b
 github.com/gohornet/hornet/plugins/tangle.configureTangleProcessor.func1(0xc00479e7e0, 0x3, 0x3, 0xc0075e2360)
/__w/hornet/hornet/plugins/tangle/tangle_processor.go:36 +0xa1
 github.com/iotaledger/hive.go/workerpool.(*WorkerPool).startWorkers.func1(0xc0003895e0)
 /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/workerpool/workerpool.go:158 +0x1b4 
 created by github.com/iotaledger/hive.go/workerpool.(*WorkerPool).startWorkers
 /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/workerpool/workerpool.go:134 +0x65 
```

**Comment by ghost** [comment]
had same  , after i    shutdown node with control  C   and  tried to connect again.   it came up with the first new milestone he did see.

windows 10

**Comment by muXxer** [comment]
Revalidation was running before, right?

**Comment by uwec** [comment]
HORNET version HORNET-0.5.0-rc7_Windows_x86_64

Have the same issue. No revalidation of database. Stopped hornet, deleted database (due to size ~28GB) - but kept my local snapshot file-  and started hornet again.  During syncing following error occured: 

```
2020-08-04T21:26:49+02:00       INFO    Tangle  Valid milestone detected! Index: 648340, Hash: XHVTK9QEM99IQBDNLHTAEASGRGYIOGIM9DJOFMFDVKFHQSCVHRZABGBVPRSSVWXZBGPTZJNYV9KTMZ999
2020-08-04T21:26:49+02:00       INFO    Tangle  Valid milestone detected! Index: 648350, Hash: HAXQTQFZA9CQXFSOAIPOSHDGNVDEOLHXKIRIZ9LAEDBZV9RAZBEBYKIYECAQYUEZKSUVIQSNNPDTXC999
2020-08-04T21:26:49+02:00       INFO    Tangle  Solidifier finished: txs: 450, collect: 42ms, total: 132ms
req(qu/pe/proc/lat): 05790/00034/02547/0561ms, reqQMs: 648322, processor: 02569, LSMI/LMI: 648279/648710, TPS (in/new/out): 06147/05388/00269, Tips (non-/semi-lazy): 0/0
req(qu/pe/proc/lat): 00000/01775/06596/2319ms, reqQMs: 0, processor: 06847, LSMI/LMI: 648279/648710, TPS (in/new/out): 06147/05388/00269, Tips (non-/semi-lazy): 0/0
2020-08-04T21:26:51+02:00       INFO    Tangle  Valid milestone detected! Index: 648318, Hash: AIZNUYRWVJKOZEYL9FQYC9VYSABOAR9GWIOODJHTMIUONLWOWPYFLBSET9IIQQVXMEPUSMKPYPIHK9999
panic: Milestone should never be updated: AIZNUYRWVJKOZEYL9FQYC9VYSABOAR9GWIOODJHTMIUONLWOWPYFLBSET9IIQQVXMEPUSMKPYPIHK9999 (648318)

goroutine 360 [running]:
github.com/gohornet/hornet/pkg/model/tangle.(*Milestone).Update(0xc01d630480, 0x2232860, 0xc01d178f60)
        /__w/hornet/hornet/pkg/model/tangle/milestones_storage.go:66 +0x11e
github.com/iotaledger/hive.go/objectstorage.(*CachedObjectImpl).updateResult(0xc001ea9580, 0x2232860, 0xc01d178f60)
        /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/objectstorage/cached_object.go:162 +0x210
github.com/iotaledger/hive.go/objectstorage.(*ObjectStorage).putObjectInCache(0xc000312680, 0x2232860, 0xc01d178f60, 0x0)
        /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/objectstorage/object_storage.go:730 +0xb3
github.com/iotaledger/hive.go/objectstorage.(*ObjectStorage).Store(0xc000312680, 0x2232860, 0xc01d178f60, 0xc018a59bc0, 0xa1a549)
        /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/objectstorage/object_storage.go:70 +0xb3
github.com/gohornet/hornet/pkg/model/tangle.StoreMilestone(0xc006a4b7c0, 0x2232701)
        /__w/hornet/hornet/pkg/model/tangle/milestones_storage.go:146 +0xdc
github.com/gohornet/hornet/pkg/model/tangle.tryConstructBundle(0xc0158c4040, 0xc0158c4000)
        /__w/hornet/hornet/pkg/model/tangle/bundle_storage.go:402 +0x95e
github.com/gohornet/hornet/pkg/model/tangle.AddTransactionToStorage(0xc00cd7fe60, 0x1010009e606, 0x1, 0x7eb343fff3e79)
        /__w/hornet/hornet/pkg/model/tangle/bundle_storage.go:304 +0x7a7
github.com/gohornet/hornet/plugins/tangle.processIncomingTx(0xc00cd7fe60, 0xc00597b270, 0xc0000befd0)
        /__w/hornet/hornet/plugins/tangle/tangle_processor.go:129 +0x9b
github.com/gohornet/hornet/plugins/tangle.configureTangleProcessor.func1(0xc029da6690, 0x3, 0x3, 0xc00a2ace40)
        /__w/hornet/hornet/plugins/tangle/tangle_processor.go:36 +0xa1
github.com/iotaledger/hive.go/workerpool.(*WorkerPool).startWorkers.func1(0xc000070460)
        /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/workerpool/workerpool.go:158 +0x1b4
created by github.com/iotaledger/hive.go/workerpool.(*WorkerPool).startWorkers
        /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/workerpool/workerpool.go:134 +0x65
```

**Comment by muXxer** [comment]
@uwec do you have a full log where the same milestone was recognized the first time?

**Comment by uwec** [comment]
@muXxer unfortunately not. I try to replicate the error, logging this time the cmd output. I also observed, that with the big database size the i/o of the ssd (connetced via USB3.0) was pretty busy (~85-90%), while with the new database is it around 1%. It also could be due to higher TPS yesterday.
I want to let it run, until database size is around 28 GB as the last time, and try to reporduce the event as above mentioned.

**Comment by ghost** [comment]
don't think it was db size... i  just synced a new node ..  turn off  node..  restart and i had the error.

**Comment by ghost** [comment]
https://paste.ofcode.org/E2zATzsVHY27JxCgTb3XSP

log . same issue

**Comment by muXxer** [comment]
Should be fixed with #593
